src/material/slider/slider-input.ts

```diff
@@ -96,7 +96,10 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
     return numberAttribute(this._hostElement.value, 0);
   }
   set value(value: number) {
-    value = isNaN(value) ? 0 : value;
+    value = value && isNaN(value) ? 0 : value;
+    if (value === null || value === undefined) {
+      value = this._getDefaultValue();
+    }
```

Comment:
i think it would be more readable if we do the `null` and `undefined` checks first and then overwrite the value if it's `NaN`. This results in a simpler diff.

```suggestion
    if (value === null || value === undefined) {
      value = this._getDefaultValue();
    }
    value = isNaN(value) ? 0 : value;
```

Reply:
Done, however one test seems to be failing not sure why. I might have missed some edge case I think so

Reply:
I have removed `undefined` check and it seems test cases failed if I checked for `undefined` may be that was some edge where we want to set `undefined`. 
If not then I think we need to update test cases where we set `undefined` to `endVal` in one of the test cases.